quickfixj-core/src/main/java/quickfix/Message.java

```diff
@@ -770,7 +770,7 @@ private void parseGroup(String msgType, StringField field, DataDictionary dd, Da
                 }
             } else {
                 // QFJ-169/QFJ-791: handle unknown repeating group fields in the body
-                if (!isTrailerField(tag) && !(DataDictionary.HEADER_ID.equals(msgType))) {
+                if (!isTrailerField(tag) && !(DataDictionary.HEADER_ID.equals(msgType) || isHeaderField(tag))) {
```

Comment:
@jacques0803 sorry for the delay. Doesn't it make sense to use `isHeaderField()` as follows:

```suggestion
                if (!isTrailerField(tag) && !(DataDictionary.HEADER_ID.equals(msgType) || isHeaderField(field, dd))) {
```

This is the same that you do above in line 681:
```
            if (isHeaderField(field, sessionDataDictionary)) {
```

Reply:
@chrjohn 
Yes it does, thanks for spotting it.  I've committed your suggestion, can you please approve the workflows again.
Regards